### PR TITLE
Add gptj notebooks

### PR DESCRIPTION
**Pull request by payoto**
Adds two GPT-J notebooks:
- generative inference
- Finetuning on the MNLI dataset

These notebooks have cached executables but do not have cached checkpoints, as the caches made the experience worse.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-data/
```
Comment (payoto) [inline .gitignore:1]: How come you are removing this?

File: .gradient/prepare-datasets.sh [hunk 2]
```diff
@@ -36,8 +36,13 @@ echo "Starting preparation of datasets"
 # symlink exe_cache files
 exe_cache_source_dir="${PUBLIC_DATASETS_DIR}/poplar-executables-pytorch-3-1-1"
 symlink-public-resources "${exe_cache_source_dir}" $POPLAR_EXECUTABLE_CACHE_DIR
+
+gptj_cache_source_dir="${PUBLIC_DATASETS_DIR}/poplar-executables-3-1-gptj"
+symlink-public-resources "${gptj_cache_source_dir}" "$POPLAR_EXECUTABLE_CACHE_DIR/gptj"
 # Symlink squad
 symlink-public-resources "${PUBLIC_DATASETS_DIR}/squad" "${HF_DATASETS_CACHE}/squad"
+symlink-public-resources "${PUBLIC_DATASETS_DIR}/glue" "${HF_DATASETS_CACHE}/glue"
+symlink-public-resources "${PUBLIC_DATASETS_DIR}/gptj-6b-checkpoints" "${CHECKPOINT_DIR}"
```
Comment (payoto) [inline .gradient/prepare-datasets.sh:45]: Checkpoints managed by transformers go to the automatically set transformers directory:

```suggestion
symlink-public-resources "${PUBLIC_DATASETS_DIR}/gptj-6b-checkpoints" "${TRANSFORMERS_CACHE}"
```

**Comment by nmb-paperspace** [comment]
Looks good!

Minor comments:

- In each notebook, make prominent at the start that an IPU-POD16 (or Bow-POD16?) is needed (it's mentioned but hidden in code cell 2 lower down)
- In the fine-tuning notebook, mention it that it takes several hours to do the fine-tuning (i.e., 20-22s * 400, which is why there's the option to just evaluate the existing checkpoint)
- Presumably also update the README_first.ipynb to include these 2 notebooks

**Review by nmb-paperspace — APPROVED** [review]

**Comment by payoto** [comment]
Thanks for review @nmb-paperspace , I added your suggestions.
